index.js

```diff
@@ -168,7 +177,36 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async submitPassword(password) {
+    // If a user is trying to submit a password, we should lock
+    // in the event that the password is incorrect
+    this.setLocked();
```

Comment:
(could) document (with a comment or in jsdoc)  the fact that setLocked can have no asynchronous side-effects. We don't want to end up running unlockKeyrings and those sideeffects in non-deterministic order.